cyral/internal/policyv2/model.go

```diff
@@ -0,0 +1,149 @@
+package policyv2
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ChangeInfo represents information about changes to the policy
+type ChangeInfo struct {
+	Actor     string `json:"actor,omitempty"`
+	ActorType string `json:"actorType,omitempty"`
+	Timestamp string `json:"timestamp,omitempty"`
+}
+
+// PolicyV2 represents the top-level policy structure
+type PolicyV2 struct {
+	Policy Policy `json:"policy,omitempty"`
+}
+
+type Scope struct {
+	RepoIds []string `json:"repoIds,omitempty"`
+}
+
+// Policy represents the policy details
+type Policy struct {
+	ID          string     `json:"id,omitempty"`
+	Name        string     `json:"name,omitempty"`
+	Description string     `json:"description,omitempty"`
+	Enabled     bool       `json:"enabled,omitempty"`
+	Scope       *Scope     `json:"scope,omitempty"`
+	Tags        []string   `json:"tags,omitempty"`
+	ValidFrom   string     `json:"validFrom,omitempty"`
+	ValidUntil  string     `json:"validUntil,omitempty"`
+	Document    string     `json:"document,omitempty"`
+	LastUpdated ChangeInfo `json:"lastUpdated,omitempty"`
+	Created     ChangeInfo `json:"created,omitempty"`
+	Enforced    bool       `json:"enforced,omitempty"`
+	Type        string     `json:"type,omitempty"`
+}
+
+// WriteToSchema writes the policy data to the schema
+func (r PolicyV2) WriteToSchema(d *schema.ResourceData) error {
+	if err := d.Set("id", r.Policy.ID); err != nil {
+		return fmt.Errorf("error setting 'id' field: %w", err)
+	}
+	if err := d.Set("name", r.Policy.Name); err != nil {
+		return fmt.Errorf("error setting 'name' field: %w", err)
+	}
+	if err := d.Set("description", r.Policy.Description); err != nil {
+		return fmt.Errorf("error setting 'description' field: %w", err)
+	}
+	if err := d.Set("enabled", r.Policy.Enabled); err != nil {
+		return fmt.Errorf("error setting 'enabled' field: %w", err)
+	}
+	if err := d.Set("tags", r.Policy.Tags); err != nil {
+		return fmt.Errorf("error setting 'tags' field: %w", err)
+	}
+	if err := d.Set("valid_from", r.Policy.ValidFrom); err != nil {
+		return fmt.Errorf("error setting 'valid_from' field: %w", err)
+	}
+	if err := d.Set("valid_until", r.Policy.ValidUntil); err != nil {
+		return fmt.Errorf("error setting 'valid_until' field: %w", err)
+	}
+
+	if err := d.Set("document", r.Policy.Document); err != nil {
+		return fmt.Errorf("error setting 'document' field: %w", err)
+	}
+
+	if err := d.Set("last_updated", map[string]interface{}{
+		"actor":      r.Policy.LastUpdated.Actor,
+		"actor_type": r.Policy.LastUpdated.ActorType,
+		"timestamp":  r.Policy.LastUpdated.Timestamp,
```

Comment:
Teach `ChangeInfo` to perform this operation, then you will call `r.Policy.LastUpdated.ToMap()` and `r.Policy.Created.ToMap()` on line `77`.

Reply:
fixed